octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/entitydetail/field/FieldEditorFactory.java

```diff
@@ -12,25 +12,58 @@
  ******************************************************************************/
 package com.hpe.octane.ideplugins.eclipse.ui.entitydetail.field;
 
+import java.util.Collection;
+import java.util.stream.Collectors;
+
 import org.eclipse.core.runtime.ILog;
 import org.eclipse.core.runtime.IStatus;
 import org.eclipse.core.runtime.Status;
+import org.eclipse.jface.viewers.LabelProvider;
 import org.eclipse.swt.SWT;
 import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
 
 import com.hpe.adm.nga.sdk.metadata.FieldMetadata;
+import com.hpe.adm.nga.sdk.metadata.FieldMetadata.Target;
 import com.hpe.adm.nga.sdk.model.EntityModel;
+import com.hpe.adm.nga.sdk.query.Query;
+import com.hpe.adm.nga.sdk.query.Query.QueryBuilder;
+import com.hpe.adm.nga.sdk.query.QueryMethod;
+import com.hpe.adm.octane.ideplugins.services.EntityService;
 import com.hpe.adm.octane.ideplugins.services.MetadataService;
 import com.hpe.adm.octane.ideplugins.services.filtering.Entity;
+import com.hpe.adm.octane.ideplugins.services.util.Util;
 import com.hpe.octane.ideplugins.eclipse.Activator;
 import com.hpe.octane.ideplugins.eclipse.ui.entitydetail.model.EntityModelWrapper;
+import com.hpe.octane.ideplugins.eclipse.util.EntityFieldsConstants;
 
 public class FieldEditorFactory {
 
+    private static final int COMBO_BOX_ENTITY_LIMIT = 100;
+
+    private static final LabelProvider DEFAULT_ENTITY_LABEL_PROVIDER = new LabelProvider() {
+        @Override
+        public String getText(Object element) {
+
+            EntityModel entityModel = (EntityModel) element;

```

Comment:
@atothhpe Can element be null or of type different than EntityModel? Would it make sense to be proactive here and check for these cases?

Reply:
I can typecheck, but it would be impossible for an element in an EntityComboBox to be null or another type, if you tried using the label provider anywhere else, it can happen.